subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -334,27 +335,28 @@ static void gen_prov_cont(struct prov_rx *rx, struct net_buf_simple *buf)
 		return;
 	}
 
-	if (rx->xact_id != link.rx.id) {
+	if (!link.rx.seg &&
+		rx->xact_id == next_transaction_id(link.rx.id)) {
```

Comment:
Misleading indentation. This line should be aligned on the with the content of the if-statement.